synapse/lib/ast.py

```diff
@@ -466,8 +466,21 @@ async def run(self, runt, genr):
             results = await stack.enter_async_context(await s_spooled.Set.anit(dirn=core.dirn, cell=core))
             revpivs = await stack.enter_async_context(await s_spooled.Dict.anit(dirn=core.dirn, cell=core))
 
-            # load the existing graph as already done
-            [await results.add(s_common.uhex(b)) for b in existing]
+            rev = {}
+
+            for b in existing:
+                await asyncio.sleep(0)
+
+                buid = s_common.uhex(b)
+                # load the existing graph as already done
+                await results.add(buid)
+
+                async for n2buid, verb in runt.snap.iterEdgePref(buid):
```

Comment:
If two nodes are specified as inputs, can we assume they have already been yielded with the intersection of edges among them?

Reply:
We should be able to yea. We can avoid make that defaultdict then.